### ISSUE TITLE
SC2320 (warning): This $? refers to echo/printf, not a previous command.

### ISSUE THREAD

**Issue by macalixx** [koalaman/shellcheck#3083]
#### For bugs
- Rule Id: SC2320
- My shellcheck version: 0.9.0
- [V] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [V] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [V] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [V] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

echo "hola" >> /tmp/prueba
Codigo=$?

echo $Codigo

```

#### Here's what shellcheck currently says:

[Line 4:](javascript:setPosition(4, 8))
Codigo=$?
       ^-- [SC2320](https://www.shellcheck.net/wiki/SC2320) (warning): This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

#### Here's what I wanted or expected to see:

An "echo" redirected to a file is a file write operation. It is necessary to obtain the exit code of that operation, because  writing to a file will fail due to lack of permissions, and sometimes it is necessary to verify it.

**Comment by brother** [comment]
Intention is very hard to catch. 
I would assume that this is a common error in the wild and that's the reason for having the warning to begin with. And if found it is easy to ignore with a directive as suggested in the wiki.

A patch for making the discovery more robust and to avoid some of these fasle positives is as always welcome.
